api/app/db/crud/hfi_calc.py

```diff
@@ -77,14 +77,62 @@ def get_most_recent_updated_hfi_request_for_current_date(session: Session,
 
 def store_hfi_request(session: Session, hfi_result_request: HFIResultRequest, username: str):
     """ Store the supplied hfi request """
-    hfi_request = HFIRequest(
+    previous_latest_request = get_most_recent_updated_hfi_request(
+        session,
         fire_centre_id=hfi_result_request.selected_fire_center_id,
-        prep_start_day=hfi_result_request.date_range.start_date,
-        prep_end_day=hfi_result_request.date_range.end_date,
-        create_timestamp=get_utc_now(),
-        create_user=username,
-        request=hfi_result_request.json())
-    session.add(hfi_request)
+        date_range=hfi_result_request.date_range)
+    stmt = (
+        insert(HFIRequest).values(fire_centre_id=hfi_result_request.selected_fire_center_id,
+                                  prep_start_day=hfi_result_request.date_range.start_date,
+                                  prep_end_day=hfi_result_request.date_range.end_date,
+                                  create_timestamp=get_utc_now(),
+                                  create_user=username,
+                                  request=hfi_result_request.json()).returning(HFIRequest.id)
+    )
+    res = session.execute(stmt).fetchone()
+    latest_hfi_request_id = res.id
+    now = get_utc_now()
+
+    if previous_latest_request is not None:
+        # Copy over ready records for existing hfi request
+        latest_hfi_ready_records = get_latest_hfi_ready_records(session, previous_latest_request.id)
+        updated_hfi_ready_records = []
+        for latest_hfi_ready_record in latest_hfi_ready_records:
+            updated_hfi_ready_records.append(HFIReady(hfi_request_id=latest_hfi_request_id,
+                                                      planning_area_id=latest_hfi_ready_record.planning_area_id,
+                                                      ready=latest_hfi_ready_record.ready,
```

Comment:
If the request is being updated - should the ready state not be set to false depending on which planning area is being updated?

Reply:
I'm not sure to be honest so I chose to keep the behaviour that adds less complexity.

Reply:
No worries - I'm sure it will surface as a bug once we start toggling values on/off.

Reply:
Actually, good call, it was in additional context: "logic: if a user changes fire starts > marks as ready > then excludes a station, changes a fire start, or a fuel type > we switch toggle back to 'not ready'"

I'll make the change

Reply:
Done